test/unit/utils/page.spec.ts

```diff
@@ -42,107 +52,162 @@ describe('Page Utils', () => {
     expect(getPage(iframe2.contentWindow)).to.be.eql('https://liveintent.com/about?key=value')
   })
 
-  it('getPage should return the url when the window location is not defined', function () {
+  it('getPage should return the url when the window location is not defined', () => {
     const iframe = document.createElement('iframe')
     iframe.src = 'https://nested.liveintent.com/about?key=value'
     document.documentElement.appendChild(iframe)
-    definedProperty(window, 'location', () => { return undefined })
 
-    expect(getPage(iframe.contentWindow)).to.be.eql('https://liveintent.com/about?key=value')
+    const oldWindow = window
+    const win: Window = {
+      ...window,
+      location: undefined
+    }
+
+    const fw: Window = {
+      ...iframe.contentWindow,
+      parent: win
+    }
+
+    // eslint-disable-next-line no-global-assign
+    window = win
+
+    expect(getPage(fw)).to.be.eql('https://liveintent.com/about?key=value')
+    // eslint-disable-next-line no-global-assign
+    window = oldWindow
```

Comment:
Should a try with resources that guarantees that the window is restored to its old state be used here? Something like this: https://github.com/LiveIntent/live-connect/pull/116/files#diff-b1c4e1833e06200917cbcbfcd57a775db133813709cf27170f2325bc45a4bcf9?

Reply:
This is a test. If it fails, then the code is wrong. I actually want it to fail in most spectacular way. That's why I don't want to put `try-finally` in this particular case.

Reply:
Are the changes to `window` visible outside of that test -e.g. if the expectation is not fulfilled and the test fails? If yes, I would rather prefer to keep it local to that test.

Reply:
Again, **if** the `expect` fails and the code never gets to `window = oldWindow`, which will lead to other tests failing - I don't really mind. IMHO, the tests should work in optimistic scenario and fail in any other way.

Reply:
I think tests should be isolated so that it is easy to understand what is broken. A test having effect on other tests - even if it is only in case of failure - is usually not what you want.

Reply:
Ok. Will fix that....

Reply:
Actually, it does not affect other tests.
Tested with failing expectation `expect(getPage(fw)).to.be.eql('https://liveintent.com/about1?key=value')` - only 1 failed test

Reply:
My point is that if it is used by other tests, it should be restored to its original state before the test ends. Tests should have no effect on each other and should not run using each other's state. I do not want to discuss it for too long. I leave it up to you if you change it or not.